docs/element-definitions.md

```diff
@@ -1,1419 +1,1623 @@
+          "content": {
+            "key": {
+              "element": "string",
+              "content": "bar"
+            },
+            "value": {
+              "element": "number"
+            }
+          }
+        }
+      ]
+    },
+    {
+      "element": "object",
+      "content": [
+        {
+          "element": "member"
+          "content": {
+            "key": {
+              "element": "string",
+              "content": "baz"
+            },
+            "value": {
+              "element": "boolean"
+            }
+          }
+        }
+      ]
+    }
+  ]
+}</pre>
+      </td>
+      <td>
+        <pre>{}</pre>
+        <pre>{
+  "foo": "Hey!"
+}</pre>
+        <pre>{
+  "bar": 42.3
+}</pre>
+        <pre>{
+  "foo": "Hey!",
+  "bar": 42.3,
+  "baz": true
+}</pre>
+      </td>
+    </tr>
+  </tbody>
+  <thead>
+    <tr>
+      <th colspan=3>Others</th>
+    </tr>
+  </thead>
+  <tbody>
+    <tr>
+      <td><a href="#ref-element">ref</a></td>
+      <td>
+        Given an named type such as
+        <pre>{
+  "element": "string",
+  "meta": {
+    "id": "SpecialString"
+  }
+}</pre>
+A reference:
+<pre>{
+  "element": "ref"
+  "content": "MyString"
+}</pre>
+      </td>
+      <td>
+        <pre>""</pre>
+        <pre>"abc"</pre>
+        <pre>"Hello world!"</pre>
+      </td>
+    </tr>
+  </tbody>
+</table>
 
-The value of the `Href` type  SHOULD be resolved as a URI-Reference per [RFC 3986][] and MAY be a relative reference to a URI.
-The value of the `Href` type MUST NOT be a URI Template.
+---
 
-### Templated Href ([String Element][])
+### Null Element
+Type with domain of a single value. Called `unit` or `()` in some programming languages.
+Note that both `nullptr` and `void` from C-style languages are unrelated concepts.
 
-The value of `Templated Href` type is to be used as a URI Template, as defined in [RFC 6570][].
-The value of the `Templated Href` type is a template used to determine the target URI of the related resource or transition.
-The value of the `Templated Href` type SHOULD be resolved as a URI-Reference per [RFC 3986][] and MAY be a relative reference to a URI.
+#### Template
+- `element` - `"null"`
 
-### Href Variables (Object Type)
+#### Example
 
-The definition is a Data Structure element `Object Type` where keys are respective URI Template variables.
+The example below defines an Element matching only the null value.
+```json
+{
+  "element": "null"
+}
+```
 
-#### Properties
+---
 
-- `element`: hrefVariables (string, fixed)
+### Boolean Element
 
-### Data Structure (Base API Element)
+Type with domain of two values: true and false.
+`content` property MUST contain a boolean value if set.
 
-Data structure definition using Data Structure elements.
+#### Template
+- `element` - `"boolean"`
+- `attributes`
+  - `typeAttributes` ([Array](#array-element)[[String](#string-element)])
+    - `fixed` ([String](#string-element)) - domain restricted to value given in `content`
```

Comment:
The type this element describes is restricted to value given in `content`. (Everywhere else where you use this sentence for primitives)